bitcoin/src/blockdata/constants.rs

```diff
@@ -46,7 +46,8 @@ pub const MAX_WITNESS_SCRIPT_SIZE: usize = 10_000;
 pub const MAX_STACK_ELEMENT_SIZE: usize = 520;
 /// How may blocks between halvings.
 pub const SUBSIDY_HALVING_INTERVAL: u32 = 210_000;
-/// Maximum allowed value for an integer in Script.
+#[deprecated = "The constant has ambiguous semantic, define your own."]
+#[doc(hidden)]
```

Comment:
Please don't hide it, that's just confusing. The deprecation message will be shown in the doc so there's no risk of people using it accidentally.

Reply:
Also I think we should add `since = "TBD"`

Reply:
It makes warning if remove original comment without hidden doc. Should I leave the comment as it was?

Reply:
Yes, please do.

Also, I would change the deprecation message to `This constant has ambiguous semantics. Please carefully check your intended use-case and define a new constant reflecting that.'